src/mqtt/qmqtt_client_p.cpp

```diff
@@ -378,17 +416,29 @@ void QMQTT::ClientPrivate::onNetworkReceived(const QMQTT::Frame& frm)
         mid = frame.readInt();
         handlePuback(type, mid);
         break;
-    case SUBACK:
+    case SUBACK: {
         mid = frame.readInt();
-        topic = _midToTopic.take(mid);
-        qos = frame.readChar();
-        handleSuback(topic, qos);
+        QMap<QString,quint8> topicMap = _midToTopic.take(mid);
+        QMapIterator<QString, quint8> i(topicMap);
+        QMap<QString,quint8> resultMap;
+        quint8 tmpqos = 0;
+         while (i.hasNext()) {
```

Comment:
wrong `while` indentation

Reply:
The rest of the code uses for-loops for iteration over QT collections. I would do that here as well, for consistency.